tools/fork-network/src/storage_mutator.rs

```diff
@@ -1,71 +1,256 @@
-use crate::single_shard_storage_mutator::SingleShardStorageMutator;
+use near_chain::types::RuntimeAdapter;
 use near_crypto::PublicKey;
-use near_epoch_manager::shard_assignment::account_id_to_shard_info;
-use near_epoch_manager::EpochManagerAdapter;
 use near_primitives::account::{AccessKey, Account};
-use near_primitives::types::{AccountId, EpochId, StateRoot};
+use near_primitives::borsh;
+use near_primitives::hash::CryptoHash;
+use near_primitives::receipt::Receipt;
+use near_primitives::shard_layout::{ShardLayout, ShardUId};
+use near_primitives::trie_key::TrieKey;
+use near_primitives::types::{AccountId, BlockHeight, ShardIndex, StateRoot, StoreKey, StoreValue};
+use near_store::adapter::flat_store::FlatStoreAdapter;
+use near_store::adapter::StoreUpdateAdapter;
+use near_store::flat::{FlatStateChanges, FlatStorageStatus};
+use near_store::{DBCol, ShardTries};
 use nearcore::NightshadeRuntime;
-use std::sync::Arc;
+
+use anyhow::Context;
+use std::sync::{Arc, Mutex};
+
+struct InProgressRoot {
```

Comment:
Please comment what it is for. Maybe part of comment for ShardUpdateState::new can go here.